### ISSUE TITLE
♻️ Update Exception Handling to use new .NET 8 `IExceptionHandler`

### ISSUE THREAD

**Issue by danielmackay** [SSWConsulting/SSW.CleanArchitecture#195]
- https://anthonygiretti.com/2023/06/14/asp-net-core-8-improved-exception-handling-with-iexceptionhandler/
- https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.diagnostics.iexceptionhandler?view=aspnetcore-8.0

**Comment by danielmackay** [comment]
- https://anthonygiretti.com/2023/06/14/asp-net-core-8-improved-exception-handling-with-iexceptionhandler/
- https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.diagnostics.iexceptionhandler?view=aspnetcore-8.0

**Comment by danielmackay** [comment]
@christoment - consider adding a new rule for this also.

**Comment by christoment** [comment]
After a bit of investigation, we still need an ExceptionHandler middleware with lambda, even with the IExceptionHandler (see https://github.com/dotnet/aspnetcore/issues/50049).
tl;dr from the issue: the exception handler lambda is used as the fallback handler when all `IExceptionHandler` returned `false`

```cs
builder.Services.AddExceptionHandler<KnownExceptionsHandler>();

// ...

app.UseExceptionHandler(exceptionHandlerApp
    => exceptionHandlerApp.Run(async context =>
    {
        IResult defaultExceptionResponse = Results.Problem(statusCode: StatusCodes.Status500InternalServerError,
            type: "https://tools.ietf.org/html/rfc7231#section-6.6.1");
    
        await defaultExceptionResponse.ExecuteAsync(context);
    }));
```

As discussed with @danielmackay  I think we should shelve this PBI for now - see https://github.com/SSWConsulting/SSW.CleanArchitecture/tree/features/195-use-net8-exceptionhandler for the WIP branch.
